### PR TITLE
qa/cephfs: add session_timeout option support

### PR DESCRIPTION
**Pull request by lxbsz**
When the mds revoking the Fwbl caps, the clients need to flush
the dirty data back to the OSDs, but the flush may make the OSDs
to be overloaded and slow, which may take more than 60 seconds to
finish. Then the MDS daemons will report the WRN messages.

For the teuthology test cases, let's just increase the timeout
value to make it work.

Fixes: https://tracker.ceph.com/issues/47565
Signed-off-by: Xiubo Li <xiubli@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/cephfs/begin.yaml [hunk 1]
```diff
@@ -47,3 +47,4 @@ tasks:
         - libuuid-devel
         - xfsprogs-devel
   - ceph:
+      session_timeout: 300
```
Comment (batrick) [inline qa/cephfs/begin.yaml:50]: Let's make this an override so it's easier to apply only to a small number of tests that we expect to run into this problem. See

https://github.com/ceph/ceph/blob/473d8e9a6066edace841e7580245f624d0afaab1/qa/tasks/ceph.py#L1792

on how to access the overrides dictionary. You would do that from filesystem.py.

File: qa/tasks/ceph.py [hunk 2]
```diff
@@ -433,6 +433,11 @@ def cephfs_setup(ctx, config):
         if max_mds > 1:
             fs.set_max_mds(max_mds)
 
+        # If absent will use the default value (60 seconds)
+        session_timeout = config.get('sesstion_timeout', 60)
+        if session_timeout != 60:
+            fs.set_session_timeout(session_timeout)
```
Comment (batrick) [inline qa/tasks/ceph.py:439]:
```suggestion
        session_timeout = config.get('sesstion_timeout')
        if session_timeout is not None:
            fs.set_session_timeout(session_timeout)
```

(and move this to `cephfs/filesystem.py`)

File: qa/tasks/ceph.py [hunk 3]
```diff
@@ -1718,6 +1723,12 @@ def task(ctx, config):
             mkfs_options: [-b,size=65536,-l,logdev=/dev/sdc1]
             mount_options: [nobarrier, inode64]
 
+    To change the mdsmap's default session_timeout (60 seconds), use::
+
+        tasks:
+        - ceph:
+            session_timeout: 300
```
Comment (batrick) [inline qa/tasks/ceph.py:1730]:
```suggestion
  overrides:
    ceph:
      fs:
        session timeout: 300
```

Comment (batrick) [reply qa/tasks/ceph.py:1730]: Let's add this to the overrides for all of the multimds suites. Add a new yaml to: qa/suites/multimds/basic/overrides and the others.

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/tasks/ceph.py (hunk 2 above)
Comment (lxbsz) [reply qa/tasks/ceph.py:439]: I didn't get it here. In `filesystem.py` I added one `set_session_timeout()` method. 
Do you mean move all the 3 lines to `filesystem.py` ?

File: qa/cephfs/begin.yaml (hunk 1 above)
Comment (lxbsz) [reply qa/cephfs/begin.yaml:50]: Sure, will do it.

File: qa/tasks/ceph.py (hunk 3 above)
Comment (lxbsz) [reply qa/tasks/ceph.py:1730]: Sure.

Comment (lxbsz) [reply qa/tasks/ceph.py:1730]: BTW, do you mean should we add a `fs:` for all the other exising ones ? like the 

```
ceph:
  max_mds: 3
```
to
```
ceph:
  fs:
    max_mds: 3
```

Right ?

Comment (lxbsz) [reply qa/tasks/ceph.py:1730]: Or maybe we should add `fs:` for all the fs related options, and then just do:

```
fs = Filesystem(...)
fsconf = config.get("fs", {})
fs.configure_filesytem(fsconf)
```

Will this make sense ?

Comment (batrick) [reply qa/tasks/ceph.py:1730]: > BTW, do you mean should we add a `fs:` for all the other exising ones ? like the
> 
> ```
> ceph:
>   max_mds: 3
> ```
> 
> to
> 
> ```
> ceph:
>   fs:
>     max_mds: 3
> ```
> 
> Right ?

Yes, please!

Comment (batrick) [reply qa/tasks/ceph.py:1730]: Usually we put these in overrides: but the difference is not significant AFAIK.

File: qa/tasks/ceph.py (hunk 2 above)
Comment (batrick) [reply qa/tasks/ceph.py:439]: I mean the session tiemout should be configured in Filesystem.create if the override is set.

File: qa/tasks/ceph.py (hunk 3 above)
Comment (lxbsz) [reply qa/tasks/ceph.py:1730]: That's right.

**Comment by batrick** [comment]
jenkins test make check

**Review by batrick — APPROVED** [review]

File: qa/suites/fs/upgrade/featureful_client/old_client/overrides/multimds/no.yaml [hunk 4]
```diff
@@ -1,3 +1,4 @@
 overrides:
   ceph:
-    max_mds: 1
+    cephfs:
+      max_mds: 1
```
Comment (batrick) [inline qa/suites/fs/upgrade/featureful_client/old_client/overrides/multimds/no.yaml:4]: We're also missing this change in several other yaml fragments like `qa/cephfs/clusters/9-mds.yaml`.

**Review by batrick — CHANGES_REQUESTED** [review]
Causes failures like /ceph/teuthology-archive/pdonnell-2020-10-22_01:21:30-multimds-wip-pdonnell-testing-20201021.232150-distro-basic-smithi/5546377/teuthology.log

File: qa/suites/fs/upgrade/featureful_client/old_client/overrides/multimds/no.yaml (hunk 4 above)
Comment (lxbsz) [reply qa/suites/fs/upgrade/featureful_client/old_client/overrides/multimds/no.yaml:4]: Missed them. Fixed already. Thanks.

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
https://pulpito.ceph.com/?branch=wip-pdonnell-testing-20201024.032205

failures unrelated. Great work @lxbsz !

**Pull request merged**